### PR TITLE
Adding collision integral data for TACOT species

### PR DESCRIPTION
**Pull request by jbscoggi**
This solves #2. 

- Added dipole polarizability data for TACOT species
- Added effective electron data for TACOT species
- Added a new feature which allows species names aliases to be defined in `collisions.xml`.
- Changed defaults for neutral-neutral interactions (explicitly updated pairs that were relying on defaults)
- Added a new test to make sure that TACOT collision integrals were properly loaded

### CODE REVIEW

File: src/transport/CollisionIntegral.h [hunk 1]
```diff
@@ -172,6 +172,19 @@ class CollisionIntegral
...
+	/**
+	 * Returns the alias of this species in the collision integral database.
+	 * If there is no alias found, then the name is returned.
+	 * 
+	 * @param root Root XMLElement representing the collision integral
+	 * database.
+	 * @param species The species.
+	 * @return std::string The alias name.
+	 */
+	std::string speciesAlias(
```
Comment (BBArrosDias) [inline src/transport/CollisionIntegral.h]: Is it necessary to be a public method?

**Comment by jbscoggi** [comment]
@grgbellasvki and @BBArrosDias, I have stalled on this PR a little bit due to a rather perplexing bug.  It seems that removing a single collision pair from the collisions.xml database causes the lookup of charged interactions to fail.  This is what is causing the current set of tests to fail.  I made sure of this by starting with a clean version and simply removing a random collision pair and can see the same behavior that occurs in the tests for this PR.  Strangely enough, this behavior seems a bit random, and only happens on Mac OS X.  This leads me to think there is a memory problem that for some reason is not detected on Linux.  If either of you have any spare time, I would love to have a second pair of eyes on this to try and figure out what's going on here.

**Comment by grgbellasvki** [comment]
@jbscoggi 
I was looking into the issue and it seems that for Mac OS X, the
end() function of XML has an strange behaviour.

The "source" of the error is in CollisionPair.cpp line 207
    if (iter == mp_xml->end()) 

iter->tag() returns correctly "Q11" while mp_xml->end()->tag() returns erroneously the same Q11.
Probably the end() function has an undefined behaviour here and gives the same pointer as iter.
I printed mp_xml->end()->tag() several times in different parts of the code and it was returning random outputs, like "species" or tags of other elements.

This was just to let you know of the problem. I will try to find a solution.

**Comment by jbscoggi** [comment]
@grgbellasvki 
So I had actually found that as well but I wanted to see if you were led to the same conclusion to make sure I didn't send you down a bad path.  Glad we agree!  Indeed the problem is still a bit hidden.  I would say this behavior is more of a symptom.  

My initial thought was we were losing the pointer to the parent in the XML tree parsing process because I was actually creating the child (which recursively creates it's children) and then pushing this child onto the children vector of the parent.  When you do that, the original object is actually copied before being placed in the vector, meaning the pointer to parent for the child's children is no longer valid....  I tried to fix this with #29, but it doesn't seem to work.  That's basically where I got before giving up and asking you guys.  And at any rate, that wouldn't explain why removing an collision pair from the database causes the problem to show up... Hopefully you will be smarter than me :)

**Comment by jbscoggi** [comment]
Hey @grgbellasvki, thanks for the recommendation.  I'll take a look on the train tomorrow!

**Comment by jbscoggi** [comment]
Hi @grgbellasvki, as it turns out, you were right that sometimes by chance the end() iterator for an XmlElement's children list would reference a real XmlElement instance which was causing this very strange bug.  I'm not sure how it didn't manifest until now, but it seems to be fixed with these latest changes (pending confirmation with TravisCI).

File: src/transport/CollisionIntegral.h (hunk 1 above)
Comment (jbscoggi) [reply src/transport/CollisionIntegral.h]: Hi @BBArrosDias, it is protected.

**Comment by jbscoggi** [comment]
Everything is passing now, I am merging.

**Pull request merged**